File: trunk/pw/Graphics/Prims3/Camera.py
# -*- coding: utf-8 -*-

from SceneNode import SceneNode
from Core.Math.Vec3 import Vec3

from OpenGL.GL import *
from OpenGL.GLU import *

class Camera(SceneNode):
    def transform(self):
        glPushMatrix()
        p = self.getVar("pos",Vec3(0,0,100))
        l = self.getVar("lookAt",Vec3(0,0,0))
        u = self.getVar("up",Vec3(0,1,0))
        gluLookAt(p.x,p.y,p.z,l.x,l.y,l.z,u.x,u.y,u.z)

File: trunk/fhe5/Graphics/Prims/Camera.cpp
#include "Camera.h"
#include <fhe/math/Vec3.h>
#include <SDL/SDL_opengl.h>

namespace fhe
{
    namespace Graphics
    {
        
        FHE_ASPECT(Camera,SceneNode3);
        
        FHE_FUNC_IMPL(Camera,msg_render3)
        {
            Vec3 pos = getEntity()->getVar<Vec3>("pos",Vec3(10,10,10)),
                lookAt = getEntity()->getVar<Vec3>("lookAt",Vec3(0,0,0)),
                up = getEntity()->getVar<Vec3>("up",Vec3(0,1,0));
                
            gluLookAt(pos.x,pos.y,pos.z,lookAt.x,lookAt.y,lookAt.z,up.x,up.y,up.z);
            
            return Var();
        }
    }
}
